src/snowflake/connector/connection.py

```diff
@@ -307,6 +314,18 @@ def __init__(self, **kwargs) -> None:
                 kwargs["application"] = "streamlit"
 
         self.converter = None
+        if config_file_path is not None:
+            # Change config file path and force update cache
+            CONFIG_PARSER.file_path = config_file_path
+            CONFIG_PARSER.read_config()
+        if connection_name is not None:
+            connections = CONFIG_PARSER["connections"]
+            if connection_name not in connections:
+                raise ProgrammingError(
```

Comment:
[ProgrammingError](https://peps.python.org/pep-0249/#programmingerror) is usually for SQL errors. We may need to raise a different error here as this is pure client.

Reply:
Ahh, I missed the fact that it's not related to the database. Would [Error](https://peps.python.org/pep-0249/#error) make sense?

Reply:
I've switched to Error